src/main/java/co/cask/hydrator/plugin/batch/HTTPToHDFSAction.java

```diff
@@ -123,14 +122,13 @@ public void run(ActionContext context) throws Exception {
                 outputStream.write(bytesIn, 0, i);
               }
             } else if (config.outputFormat.equalsIgnoreCase("Text")) {
-              try (Reader reader = new InputStreamReader(inputStream, config.charset)) {
-                StringBuilder inputStringBuilder = new StringBuilder();
-                BufferedReader bufferedReader = new BufferedReader(reader);
-                String line;
-                while ((line = bufferedReader.readLine()) != null) {
-                  inputStringBuilder.append(line + "\n");
+              try (BufferedReader bufferedReader = new BufferedReader
+                (new InputStreamReader(inputStream, config.charset));) {
+                int i = 0;
+                char charsIn[] = new char[BUFFER_SIZE];
+                while ((i = bufferedReader.read(charsIn)) >= 0) {
+                  outputStream.write(new String(charsIn).getBytes(), 0, i);
```

Comment:
Other way to do convert char to byte would be as below.Please suggest
    byte[] bytes = new byte[charsIn.length];
    for (int i = 0; i < charsIn.length; i++) {
      bytes[i] = (byte) charsIn[i];
    }